core/locked_list.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <list>
+#include <mutex>
+
+namespace dronecore {
+
+// Note: this wrapper around list was developed for `MavlinkCommands` but was not used
+// eventually. Instead, `std::list` was used directly and protected with a `std::mutex`.
+
+template <class T>
+class LockedList
+{
+public:
+    LockedList() :
+        _list(),
+        _mutex()
+    {};
+
+    class iterator: public std::list<T>::iterator
+    {
+    public:
+        iterator(typename std::list<T>::iterator c, std::mutex &mutex) :
```

Comment:
I know we are using single char variables elsewhere but we should move away from this, can we name this variable something more descriptive, please?

Reply:
Ok 173ac1a.